libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/LanguageGeneratorManager.cs

```diff
@@ -68,7 +68,7 @@ public static ImportResolverDelegate ResourceExplorerResolver(string locale, Dic
                 else
                 {
                     var content = resource.ReadTextAsync().GetAwaiter().GetResult();
-                    return (content, resource.Id);
+                    return (content, (resource as FileResource).FullName);
```

Comment:
>(resource as FileResource).FullName) [](start = 37, length = 36)

I don't think this is right.  ImportResolverDelegate definition is that it returns (content, resource.id).  FullName is not the resource id

Reply:
Currently, The id of file resource is "file name", but what we really want is "file path", which can be obtained from "FullName" property.

Reply:
LG system allow an id system specified by ImportResolver, which can be fileName, or fullPath, depends on what import resolver returns. 

If i understand correctly, what @Danieladu did here,  is two things
1. fix an inconsistency that some files we are using fileName as id and some other files we are using fullPath as id. In this PR, we make sure we use fullPath all across any places
2. the reason we align all places to fullPath, is to benefit debugger registeration, right? @Danieladu? is this a must have for debugger? if it's not a must have, i think align all places to fileName may aligns better with resource explorer convention.

Reply:
Yes. Currently, the absolute path of LG file is a must for the debugger. The debugger can not register the breaking points without the absolute path.

Reply:
ImportResolverDelegate says it returns (content, RESOURCEID)  If the signature is that it returns a resource.id, then you need to return the resource.id, its not "Return anything which is an id" it's a resource id.  The problem with assuming it will be a file is that you are BREAKING for when people want to run in the cloud with a non-file system resource explorer .(Which is exactly what PowerVA wants to do)


---
In reply to: [446817914](https://github.com/microsoft/botbuilder-dotnet/pull/4155#discussion_r446817914) [](ancestors = 446817914)

Reply:
We should never be using full path as the id.  That would break the resource explorer.

---
In reply to: [446806570](https://github.com/microsoft/botbuilder-dotnet/pull/4155#discussion_r446806570) [](ancestors = 446806570)

Reply:
drop the full path as id makes sense. But now the LG debugger has a requirement, needs to dynamically obtain the fullpath of the resource in the generator to set the breaking points, and the generator cannot rely on the explorer, is there any good solution to achieve this goal? Thanks,